### PR TITLE
Refactor .po and .mo files generation

### PR DESCRIPTION
**Pull request by KJhellico**
## Proposed change

- code optimization
- additional script `msgfmt.py` removed
- `gettext` package binaries (`msgmerge`) requirement removed

## Type of change

- [ ] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [x] Existing code/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency upgrade (version update)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] This PR is filed against `beta` branch of the repository
- [x] This PR doesn't contain any merge conflicts and has clean commit history
- [x] The code style looks good (`make pre-commit`)
- [x] I've added tests to verify that the new code works and all tests pass locally (`make test`, `make tox`)
- [x] The related [documentation][docs] has been added/updated (check off the box for free if no updates is required)

[contributing-guidelines]: https://github.com/dr-prodigy/python-holidays/blob/beta/CONTRIBUTING.rst
[docs]: https://github.com/dr-prodigy/python-holidays/tree/beta/docs/source

### CODE REVIEW

File: scripts/l10n/generate_po_files.py [hunk 1]
```diff
@@ -13,31 +13,37 @@
...
         """Processes entities in specified directory."""
         country_code_info_mapping = {}
-        module_name_path_mapping = {
-            str(path).split(os.sep)[-1].replace(".py", ""): str(path)
-            for path in Path(os.path.join("holidays", "countries")).glob(
-                "*.py"
-            )
-            if not str(path).endswith("__init__.py")
-        }
+        module_name_path_list = (
```
Comment (arkid15r) [inline scripts/l10n/generate_po_files.py:36]: Would just `modules` be okay here?
```suggestion
        modules = (
```

File: scripts/l10n/generate_po_files.py [hunk 2]
```diff
@@ -52,18 +58,16 @@ def process_countries(self):
...
-                "holidays", "locale", "pot", f"{country_code}.pot"
-            )
+            pot_file_path = str(locale_path / "pot" / f"{country_code}.pot")
             # Create .pot files.
-            subprocess.run(
-                (
-                    "pot-create",
-                    class_file_path,
+            pot_create(
+                [
```
Comment (arkid15r) [inline scripts/l10n/generate_po_files.py:69]:
```suggestion
                (
```

File: scripts/l10n/generate_po_files.py [hunk 3]
```diff
@@ -74,41 +78,20 @@ def process_countries(self):
                     package_version,
                     "--width",
                     "80",
-                ),
-                check=True,
+                ],
```
Comment (arkid15r) [inline scripts/l10n/generate_po_files.py:81]:
```suggestion
                ),
```

**Review by arkid15r — COMMENTED** [review]
It's a nice way to get rid of `msgfmt.py`!

File: scripts/l10n/generate_mo_files.py [hunk 4]
```diff
@@ -11,33 +11,22 @@
...
-                (
-                    sys.executable,
-                    os.path.join("scripts", "l10n", "msgfmt.py"),
-                    "-o",
-                    mo_file,
-                    po_file,
-                ),
-                check=True,
-            )
+        for po_path in (Path() / "holidays" / "locale").rglob("*.po"):
```
[the remark this answers is not in the document]
Comment (KJhellico) [reply scripts/l10n/generate_mo_files.py:24]: `Path()` handles it internally.

File: scripts/l10n/generate_mo_files.py [hunk 5]
```diff
@@ -11,33 +11,22 @@
...
+            mo_path = po_path.with_suffix(".mo")
+            if mo_path.exists():
+                mo_path.unlink()
+            po = pofile(str(po_path))
```
[the remark this answers is not in the document]
Comment (KJhellico) [reply scripts/l10n/generate_mo_files.py]: Done, here and below.

**Comment by arkid15r** [comment]
It looks like I need to learn more about pathlib. LGTM for now, will look deeper later.

File: scripts/l10n/generate_mo_files.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (arkid15r) [reply scripts/l10n/generate_mo_files.py:24]: Got it 👍

**Pull request merged**